week4/assignment/src/components/common/Input.tsx

```diff
@@ -0,0 +1,113 @@
+import { HTMLAttributes } from "react";
+import styled from "@emotion/styled";
+import { Theme } from "../../styles/theme";
+import { useState } from "react";
+import OpenEye from "../../assets/OpenEye.svg?react"
+import CloseEye from "../../assets/CloseEye.svg?react"
+
+
+export interface InputProps extends HTMLAttributes<HTMLInputElement> {
+  type: string;
+  value: string | number;
+  name: string;
+  placeholder?: string;
+  errorMessage?: string;
+  isValid?: boolean;
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  showPwd?: boolean;
+}
```

Comment:
```suggestion
export interface InputProps extends HTMLAttributes<HTMLInputElement> {
  errorMessage?: string;
  isValid?: boolean;
  showPwd?: boolean;
}
```

여기도 Button 컴포넌트와 마찬가지로 상속받고 있으니 몇몇 속성은 빼줘도 괜찮을 것 같아요!
혹시 필수 값으로 받기위해서 명시적으로 작성하신 것이라면, optional(?)인 placeholder만 뺄 수 있겠네요!